scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamic.scala

```
@@ -241,6 +241,23 @@ final case class ScalafmtDynamic(
 
   override def withMavenRepositories(repositories: String*): Scalafmt =
     copy(repositories = repositories.map(MavenRepository.of).toList)
+
+  override def createSession(config: Path): ScalafmtSession =
+    resolveConfig(config).fold(
+      error => { reportError(config, error); null },
```

May be just throw exeption instead of `null` handling?